### DISCUSSION TITLE
centos 重启会导致 Java环境的docker容器被删除

### DISCUSSION THREAD

**Discussion by finallycc** [1Panel-dev/1Panel#6974] [Q&A]
### 1Panel 版本

v1.10.20-lts

### 问题描述

高负载后，我重启了系统，Java环境的配置都在，但是容器已经被删除。

### 重现步骤

配置Java环境后启动后，重启服务器。

### 期待的正确结果

Java环境的docker 容器在系统重启后也不会被删除，正常运行

**Comment by wanghe-fit2cloud** [comment]
在高负载后重启系统导致 Docker 容器被删除，可能由以下原因引起：

1.	Docker 配置或存储驱动问题：在高负载的情况下，Docker 可能无法正确地同步和保存容器元数据，特别是如果存储驱动不稳定。检查 /var/lib/docker 目录下的数据是否完整。
2.	意外的 Docker 重置或清理：有些系统配置或运维脚本可能会在重启时自动清理未运行的 Docker 容器。如果系统设置了自动清理机制（如定期清除停止的容器或重建环境的脚本），可能会误删正常运行的容器。
3.	文件系统损坏：在高负载时，文件系统（如 ext4、xfs）可能会出错，导致容器数据损坏或丢失。重启后，Docker 可能无法找到原有的容器数据。
4.	意外的 Docker 配置丢失：可能在 /etc/docker 或 /var/lib/docker 目录下有部分 Docker 配置文件损坏，导致 Docker 在启动时将容器误认为丢失并清理掉了残留数据。
5.	数据未正确持久化：如果 Docker 没有设置 --restart 策略来保留容器状态，系统重启时这些容器不会自动重启，甚至在某些情况下会被清理。

建议先检查 Docker 的日志文件（通常位于 /var/log/docker.log），查找重启时的具体错误信息。同时，确认 Docker 存储驱动、容器管理脚本、系统的清理机制是否有异常配置。
